### PR TITLE
Adding BucketSort and SelectionSort in JavaScript

### PR DESCRIPTION
**Pull request by rushi-173**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have added BucketSort and SelectionSort algorithms in javascript.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#2075 #
(Write your answer here.)

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@rushi-173 Please update the README file.

File: JavaScript/sort/BucketSort.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+function bucketSort(arr, bucketSize) {
```
Comment (atarax665) [inline JavaScript/sort/BucketSort.js:1]: Add a beginner pitch of the algorithm along with space and time complexity in commented form.

File: JavaScript/sort/BucketSort.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+function sortBuckets(buckets) {
+    const sortedArr = [];
+    for (let i = 0; i < buckets.length; i++) {
+        if (buckets[i] != null) {
+            insertionSort(buckets[i]); // quick sort is another good option
+            sortedArr.push(...buckets[i]);
+        }
+    }
+    return sortedArr;
+}
```
Comment (atarax665) [inline JavaScript/sort/BucketSort.js:47]: Please add driver code.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
Is it being worked on @rushi-173 ?

**Comment by HarshCasper** [comment]
Please work only on those Issues that you are assigned. The original assignee is @Sloth-Panda

**Pull request closed without merging**